src/main/resources/db/changelog/ddl/0023_alter_aed_and_aed_contact_details.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+
+    <changeSet author="vishal.gupta@valtech.com" id="0023a_alter_aed_contact_details - add sanitized_surname column and indexes" context="ddl">
+        <sql>
+            <comment>Resulted sanitized_surname is created from surname by uppercasing, replacing hyphen with space, removing any apostrophe, replacing multiple spaces to a single space</comment>
+            ALTER TABLE aed_contact_details ADD sanitized_surname VARCHAR(255) AS (TRIM(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(UPPER(surname),'-',' '),'\'',''),'[ ]{2,}',' '))) STORED AFTER surname;
+        </sql>
+
+        <createIndex tableName="aed_contact_details" indexName="aed_contact_details_sanitized_surname_idx">
+            <column name="sanitized_surname"/>
+        </createIndex>
+
+        <rollback>
+            <dropIndex tableName="aed_contact_details" indexName="aed_contact_details_sanitized_surname_idx"/>
+            <dropColumn tableName="aed_contact_details" columnName="sanitized_surname"/>
+        </rollback>
+    </changeSet>
+
+    <changeSet author="vishal.gupta@valtech.com" id="0023b_alter_anonymous_elector_document - add sanitized_electoral_roll_number column and indexes" context="ddl">
+        <sql>
+            <comment>Resulted sanitized_electoral_roll_number is created from electoral_roll_number by uppercasing, replacing hyphen with space, removing any apostrophe, replacing multiple spaces to a single space</comment>
+            ALTER TABLE anonymous_elector_document ADD sanitized_electoral_roll_number VARCHAR(30) AS (TRIM(REGEXP_REPLACE(REGEXP_REPLACE(REGEXP_REPLACE(UPPER(electoral_roll_number),'-',' '),'\'',''),'[ ]{2,}',' '))) STORED AFTER electoral_roll_number;
```

Comment:
In some example data I noticed that the outcome of the sanitizing looks like:
![image](https://user-images.githubusercontent.com/56517757/233576243-750b1270-ba35-411e-9f6b-3ec96e8d3907.png)

It might be safer to remove all non alphanumerics, e.g.
```
AS (TRIM(REGEXP_REPLACE(UPPER(electoral_roll_number),'[^[:alnum:]]',''))) STORED
```

Reply:
Thanks, however I think it's better to not overcomplicate this bit for MVP if that's ok.
The regex that I used replaces `hypens` with `a space`